source/sender.ts

```diff
@@ -58,25 +62,25 @@ function manageConnection(
   type: string,
   options: Options,
   target: AnyTarget,
-  sendMessage: () => Promise<unknown>
+  sendMessage: (attempt: number) => Promise<unknown>
 ): Promise<unknown> | void {
   if (!options.isNotification) {
     return manageMessage(type, target, sendMessage);
   }
 
-  void sendMessage().catch((error: unknown) => {
+  void sendMessage(1).catch((error: unknown) => {
     debug(type, "notification failed", { error });
   });
 }
 
 async function manageMessage(
   type: string,
   target: AnyTarget,
-  sendMessage: () => Promise<unknown>
+  sendMessage: (attempt: number) => Promise<unknown>
 ): Promise<unknown> {
   const response = await pRetry(
```

Comment:
Side note: p-retry’s stack traces are not great because they're handled via `setTimeout`. I might replace it with a plain `while(retries--) await sleep()` loop.